inject.go

```diff
@@ -183,7 +183,7 @@ func (i *injector) Get(t reflect.Type) reflect.Value {
 		}
 	}
 	if len(impls) > 1 && i.options.PanicOnAmbiguity {
-		panic(fmt.Sprintf("Expect single matching implementation but found %v", len(impls)))
+		panic(fmt.Sprintf("Expected single matching implementation for type <%v> but found %v: %v", t, len(impls), impls))
```

Comment:
Does reflect.Value stringify nicely for all types? The documentation for reflect.Value's String() method (https://golang.org/pkg/reflect/#Value.String) says "The fmt package treats Values specially. It does not call their String method implicitly but instead prints the concrete values they hold.", but your test seems to indicate it prints the type, so I don't know what to believe anymore.